diff --git a/terms.c b/terms.c
index b0a15f7..a883957 100644
--- a/terms.c
+++ b/terms.c
@@ -1678,7 +1678,46 @@ static char *
 color_seq(int colmode)
 {
     static char seqbuf[32];
-    sprintf(seqbuf, "\033[%dm", ((colmode >> 8) & 7) + (highIntensityColors ? 90 : 30));
+		 int color = ((colmode >> 8) & 7);
+		 switch(color)  {
+			case 1:
+				// WHITE
+				sprintf(seqbuf, "\x1b[38;5;23m");
+				break;
+			case 2:
+				// RED
+				sprintf(seqbuf, "\x1b[38;5;203m");
+				break;
+			case 3:
+				// GREEN
+				sprintf(seqbuf, "\x1b[38;5;28m");
+				break;
+			case 4:
+				// BLUE
+				sprintf(seqbuf, "\x1b[38;5;33m");
+				break;
+			case 5:
+				// 
+				sprintf(seqbuf, "\x1b[38;5;62m");
+				break;
+			case 6:
+				// CYAN
+				sprintf(seqbuf, "\x1b[38;5;37m");
+				break;
+			case 7:
+				// WHITE
+				sprintf(seqbuf, "\x1b[38;5;23m");
+				break;
+			default:
+				sprintf(seqbuf, "\x1b[38;5;23m");
+				/* sprintf(seqbuf, "\033[%dm", ((colmode >> 8) & 7) + 30); */
+		 }
+		 /* if (color == 1) { */
+			/* sprintf(seqbuf, "\x1b[38;5;23m"); */
+		 /* } else { */
+			/* sprintf(seqbuf, "\033[%dm", ((colmode >> 8) & 7) + 30); */
+		 /* } */
+		/* sprintf(seqbuf, "\x1b[38;5;%dm", ((colmode >> 8) & 7) + 22); */
     return seqbuf;
 }
 
@@ -1695,7 +1734,41 @@ static char *
 bcolor_seq(int colmode)
 {
     static char seqbuf[32];
-    sprintf(seqbuf, "\033[%dm", ((colmode >> 12) & 7) + 40);
+		 /* sprintf(seqbuf, "\033[%dm", ((colmode >> 12) & 7) + 40); */
+		int color = ((colmode >> 12) & 7);
+		switch(color)  {
+		 case 1:
+			 // WHITE
+			 sprintf(seqbuf, "\x1b[38;5;23m");
+			 break;
+		 case 2:
+			 // RED
+			 sprintf(seqbuf, "\x1b[38;5;203m");
+			 break;
+		 case 3:
+			 // GREEN
+			 sprintf(seqbuf, "\x1b[38;5;28m");
+			 break;
+		 case 4:
+			 // BLUE
+			 sprintf(seqbuf, "\x1b[38;5;33m");
+			 break;
+		 case 5:
+			 // 
+			 sprintf(seqbuf, "\x1b[38;5;62m");
+			 break;
+		 case 6:
+			 // CYAN
+			 sprintf(seqbuf, "\x1b[38;5;37m");
+			 break;
+		 case 7:
+			 sprintf(seqbuf, "\x1b[38;5;122m");
+			 break;
+		 default:
+			 sprintf(seqbuf, "\x1b[38;5;23m");
+			 /* sprintf(seqbuf, "\033[%dm", ((colmode >> 8) & 7) + 30); */
+		}
+		/* sprintf(seqbuf, "\x1b[38;5;%dm", ((colmode >> 12) & 7) + 40); */
     return seqbuf;
 }
 #endif				/* USE_BG_COLOR */
